doc/source/data/api/dataset.rst

```diff
@@ -0,0 +1,285 @@
+.. _dataset-api:
+
+Dataset API
+===========
+
+.. autoclass:: ray.data.Dataset
+
+**Basic Transformations**
+
+.. autosummary::
+    :nosignatures:
+
+    ray.data.Dataset.map
```

Comment:
A stretch request: add a ref to each API, so that we can pinpoint to individual API, which is useful. I think what's needed is adding something like ".. _dataset-map-ref:".

Reply:
This should already be natively supported! 

```
:py:meth:`ray.data.Dataset.map`
```

Reply:
@jianoaix For the individual APIs, that should already be doable with cross-referencing via e.g. `:meth:``ray.data.Dataset.map```

Reply:
OK, can it be linked with a url? e.g. https://docs.ray.io/en/master/data/package-ref.html#dataset-api, if we point a user to map(), can we give a ulr pointing to it?

Reply:
Found it: https://docs.ray.io/en/master/data/package-ref.html#ray.data.Dataset.map. I guess I was looking at the navigation list on the right-hand-side which doesn't show it.